### ISSUE TITLE
Kernel dead when running a second time plotter.add_mesh_slice with Plotter(notebook=False)

### ISSUE THREAD

**Issue by gbene** [pyvista/pyvista#2714]
### Describe the bug, what's wrong, and what you expected.

Hi!

I was trying the slicing features (in particular add_mesh_slice) in jupyter notebook using the knee_full example. When plotting in cell I'm not able to interact with the slice plane. I can however use the notebook=False option in Plotter to open a separate window. When I do this a couple of strange things happen. The first time i run the cell it works (I can use the widget) but when I close the plot window (both with the top right x or by pressing q) a second frozen plot window appears. One of two things happens:

1. If I close this window and run the cell again a black window appears and after a short amount of time the kernel crashes. 

2. If I don't close the frozen window I can run the cell again and works as it should but when I close it again another frozen window appears.

### Steps to reproduce the bug.

cell 1:
```python 
import pyvista as pv
from pyvista import examples
 ```
cell 2:
 ```python
plotter =pv.Plotter(notebook=False)

plotter.add_mesh_slice(vol)
plotter.show()
``` 

When running cell 2 the plot window appears.

![image](https://user-images.githubusercontent.com/40857804/170867187-cfb7358f-19dd-45f6-b32e-daaf790c8afc.png)
 

When closed a second frozen window will appear

![image](https://user-images.githubusercontent.com/40857804/170867216-3f148216-f7b8-49a8-9345-38b70245ddfe.png)

If the second window is not closed the plot opens correctly when the cell is run again (but the frozen window remains there) 

![image](https://user-images.githubusercontent.com/40857804/170867274-f01fbbb5-03c7-4942-addc-dbd5c8b01943.png)

If the fozen window is closed a black window appears and then the kernel dies

![image](https://user-images.githubusercontent.com/40857804/170869197-b60e564b-b056-4076-b3c1-781ee0e5315a.png)
![image](https://user-images.githubusercontent.com/40857804/170869205-e32f0634-6ec6-41c1-99e4-f42237cfe4d6.png)

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Sun May 29 13:49:19 2022 CEST

                OS : Linux
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
               RAM : 15.5 GiB
       Environment : Jupyter
       File system : ext4
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA GeForce GTX 1050/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 510.68.02

  Python 3.9.12 (main, Apr  5 2022, 06:56:58)  [GCC 7.5.0]

           pyvista : 0.34.1
               vtk : 9.1.0
             numpy : 1.22.3
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.12
        matplotlib : 3.5.1
             PyQt5 : 5.9.2
           IPython : 8.1.1
          colorcet : 1.0.0
        ipyvtklink : 0.2.2
             scipy : 1.7.3
        itkwidgets : 0.32.1
              tqdm : 4.64.0

  Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904
  for Intel(R) 64 architecture applications
--------------------------------------------------------------------------------
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by akaszynski** [comment]
Bad news: I'm able to reproduce your error with `pyvista==0.34.1`

Good news: It's been fixed as of #2569.

Install the latest with:
```
pip install git+https://github.com/pyvista/pyvista.git@main
```

Let me know if this solves your issue. We should have the next release out in a few weeks, end of June at the latest.

**Comment by gbene** [comment]
Sorry for the late response. Yes by installing the new version the problems are fixed! Thanks!
